server/src/test/java/org/elasticsearch/gateway/ReplicaShardAllocatorIT.java

```diff
@@ -41,8 +50,10 @@
 import java.util.stream.IntStream;
 
 import static org.elasticsearch.test.hamcrest.ElasticsearchAssertions.assertAcked;
+import static org.hamcrest.Matchers.empty;
 import static org.hamcrest.Matchers.equalTo;
 import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.not;
 
 @ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 0)
 public class ReplicaShardAllocatorIT extends ESIntegTestCase {
```

Comment:
I think there would be value to an `ESIntegTestCase` showing that we pick a copy suitable for ops-based recovery in a full-cluster restart in which there was ongoing indexing, as well as these tests of the behaviour of no-op recoveries.

Reply:
Yes, I add a test in https://github.com/elastic/elasticsearch/pull/46959/commits/a299c0f16a6421d0ea1b2e8481f3a1864ef320ce.